go.mod

```diff
@@ -19,13 +18,13 @@ require (
 	github.com/spf13/viper v1.13.0
 	golang.org/x/net v0.0.0-20220805013720-a33c5aa5df48
 	gopkg.in/yaml.v2 v2.4.0
-	k8s.io/api v0.22.3
-	k8s.io/apiextensions-apiserver v0.22.3
-	k8s.io/apimachinery v0.22.3
+	// NOTE: we need to stay on k8s v1.24 as installations run v1.23 apart from the deprecated kvm.
+	k8s.io/api v0.24.6
```

Comment:
Upgrading to 0.24.6 as all installations apart from kvm are on 1.23.9. This allows us to upgrade the cluster-api dependency which allows us to upgrade the controller runtime dependency.

Once this is merged, we need to create a major release as this will not work on old prometheus versions.

I would advocate that we create a branch for kvm installatoins if we ever need to fix something there